src/rewards/mod.rs

```diff
@@ -171,73 +173,87 @@ impl<'a> RewardsMonitor<'a> {
             .ok_or_else(|| anyhow!("current epoch has no rewards"))?;
 
         // Extract into staking rewards and validator rewards.
-        let staking_rewards = current_rewards
-            .into_iter()
-            .filter(|reward| reward.reward_type == Some(RewardType::Staking))
-            .collect::<Vec<_>>();
-
-        // Copy staking_rewards into rewards
-        for staking_reward in staking_rewards.iter().cloned() {
-            rewards.insert(
-                (staking_reward.pubkey.parse()?, current_epoch),
-                staking_reward.clone(),
-            );
-        }
+        let staking_rewards = current_rewards.into_iter().filter_map(|r| {
+            if r.reward_type != Some(RewardType::Staking) {
+                return None;
+            }
+
+            if let Ok(pubkey) = r.pubkey.parse() {
```

Comment:
```suggestion
                None
            }
            else if let Ok(pubkey) = r.pubkey.parse() {
```

Reply:
Possibly yes, although the `if` and `else if` branches are rather unrelated in that case.

Reply:
Either is fine, but a single if-elseif-else block returns a single type and seems a bit more idiomatic

Reply:
Fixed in a999631.